src/java/org/jivesoftware/openfire/plugin/session/RemoteClientSession.java

```diff
@@ -310,4 +336,64 @@ public void readExternal(ObjectInput in) throws IOException, ClassNotFoundExcept
             initialized = ExternalizableUtil.getInstance().readBoolean(in);
         }
     }
+
+/*
+*Addes for OF-1868 
+*/
+	private static class SetMessageCarbonsTask extends ClientSessionTask {
```

Comment:
Note sure these cluster tasks are necessary.

The only place the setters are called are from IQBlockingHandler & IQMessageCarbonsHandler. In both cases, they act on the receipt of a message from the client - so the ClientSession will always be a LocalClientSession. 

(Though I appreciate you've just copied the code from the other set* methods, I don't think any of those are necessary either)

Reply:
@guusdk - any thoughts on the above assertion? 

i.e. if an IQHandler calls `sessionManager.getSession(iq.getFrom())`  will that session always be a `LocalClientSession`?

My reasoning is that;
a) The IQHandler runs on the node that receives the packet
b) The node that receives the packet is the node to which the session is connected
c) If you're running on the node to which the client session is connected, `sessionManager.getSession()` will return a `LocalClientSession`